sacred/observers/file_storage.py

```diff
@@ -51,16 +52,48 @@ def __init__(self, basedir, resource_dir, source_dir, template,
         self.cout = ""
         self.cout_write_cursor = 0
 
-    def queued_event(self, ex_info, command, host_info, queue_time, config,
-                     meta_info, _id):
-        if not os.path.exists(self.basedir):
-            os.makedirs(self.basedir)
+    @staticmethod
+    def _makedirs(name, mode=0o777, exist_ok=False):
+        """ Wrapper of os.makedirs with fallback
+            for exist_ok on python 2.
+        """
+        try:
+            os.makedirs(name, mode, exist_ok)
+        except TypeError:
+            if not os.path.exists(name):
+                os.makedirs(name, mode)
+
+    def _try_set_next_dir(self, _id, raise_error):
```

Comment:
I would move the error handling outside of  `_try_set_next_dir`. With the current code the outer function needs to keep track of the error count, but the inner function raises the error, so you have to pass a flag. Instead I would wrap  `_try_set_next_dir` in a try-except block. Also the  `_id` argument seems to be unused, so I would remove both arguments.

Reply:
I installed your suggestion.